drivers/pds/pdsrestore/restore_utils.go

```diff
@@ -0,0 +1,160 @@
+package pdsrestore
+
+import (
+	"fmt"
+	pds "github.com/portworx/pds-api-go-client/pds/v1alpha1"
+	pdsapi "github.com/portworx/torpedo/drivers/pds/api"
+	ds "github.com/portworx/torpedo/drivers/pds/dataservice"
+	tc "github.com/portworx/torpedo/drivers/pds/targetcluster"
+	"github.com/portworx/torpedo/pkg/log"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"strings"
+	"time"
+)
+
+const (
+	restoreTimeOut         = 30 * time.Minute
+	restoreTimeInterval    = 20 * time.Second
+	restoreMaxTimeInterval = 1 * time.Minute
+)
+
+type RestoreClient struct {
+	TenantId             string
+	ProjectId            string
+	Components           *pdsapi.Components
+	Deployment           *pds.ModelsDeployment
+	RestoreTargetCluster *tc.TargetCluster
+}
+
+// DSEntity struct contain the current context of data and deployment object. Which could be later utilized as source of truth for validating the restored object
+type DSEntity struct {
+	Deployment *pds.ModelsDeployment
+	// TODO Add datahash for data validation
+}
+
+func (restoreClient *RestoreClient) TriggerAndValidateRestore(backupJobId string, namespace string, bkpDsEntity DSEntity, isRestoreInSameNS, validate bool) (*pds.ModelsRestore, error) {
+	var (
+		bkpJob                 *pds.ModelsBackupJob
+		nsName, pdsNamespaceId string
+	)
+	k8sClusterId, err := restoreClient.RestoreTargetCluster.GetClusterID()
+	if err != nil {
+		return nil, fmt.Errorf("unable to fetch the cluster Id")
+	}
+	pdsRestoreTargetClusterId, err := restoreClient.RestoreTargetCluster.GetDeploymentTargetID(k8sClusterId, restoreClient.TenantId)
+	if err != nil {
+		return nil, fmt.Errorf("unable to fetch the cluster details from the control plane")
+	}
+	if !isRestoreInSameNS {
+		nsName, pdsNamespaceId, err = restoreClient.getNameSpaceId(pdsRestoreTargetClusterId)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		bkpJob, err = restoreClient.Components.BackupJob.GetBackupJob(backupJobId)
+		if err != nil {
+			return nil, err
+		}
+		nsName, pdsNamespaceId = namespace, bkpJob.GetNamespaceId()
+	}
+	log.Infof("backup Job - %v,Restore Target Cluster Id - %v, NamespaceId - %v", backupJobId, pdsRestoreTargetClusterId, pdsNamespaceId)
+	restoredModel, err := restoreClient.Components.Restore.RestoreToNewDeployment(backupJobId, "autom", pdsRestoreTargetClusterId, pdsNamespaceId)
+	if err != nil {
+		return nil, fmt.Errorf("failed during restore")
+	}
+	err = wait.Poll(restoreTimeInterval, restoreTimeOut, func() (bool, error) {
+		restore, err := restoreClient.Components.Restore.GetRestore(restoredModel.GetId())
+		state := restore.GetStatus()
+		if err != nil {
+			log.Errorf("failed during fetching the restore object, %v", err)
```

Comment:
do not log the err if you are returning. as this be logged in calling function also